src/common/Cryptography/HMAC.h

```diff
@@ -0,0 +1,138 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TRINITY_HMAC_H
+#define TRINITY_HMAC_H
+
+#include "Define.h"
+#include "Errors.h"
+#include <array>
+#include <string>
+#include <string_view>
+#include <openssl/hmac.h>
+
+class BigNumber;
+
+namespace Trinity {
+    namespace Impl {
+        struct HMACImpl
+        {
+            typedef EVP_MD const* (*HashCreator)();
+
+#if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER < 0x10100000L
+            static HMAC_CTX* MakeCTX()
+            {
+                HMAC_CTX* ctx = new HMAC_CTX();
```

Comment:
https://www.openssl.org/docs/manmaster/man3/EVP_DigestSignInit.html

These are the corresponding `EVP_*` functions.

Reply:
Can you find any documentation on the `PKEY` functions for openssl < 1.1.1? I couldn't.

Reply:
Should be solved when we drop <1.1.1.